File: mmupnp/src/test/java/net/mm2d/upnp/internal/thread/TaskExecutorsTest.java
/*
 * Copyright (c) 2019 大前良介 (OHMAE Ryosuke)
 *
 * This software is released under the MIT License.
 * http://opensource.org/licenses/MIT
 */

package net.mm2d.upnp.internal.thread;

import net.mm2d.upnp.TaskExecutor;

import org.junit.Before;
import org.junit.Test;

import static org.mockito.Mockito.*;

public class TaskExecutorsTest {
    private TaskExecutor mCallback;
    private TaskExecutor mIo;
    private TaskExecutors mTaskExecutors;

    @Before
    public void setUp() throws Exception {
        mCallback = mock(CallbackExecutor.class);
        mIo = mock(IoExecutor.class);
        mTaskExecutors = new TaskExecutors(mCallback, mIo);
    }

    @Test
    public void callback() {
        final Runnable task = mock(Runnable.class);
        mTaskExecutors.callback(task);

        verify(mCallback, times(1)).execute(task);
    }

    @Test
    public void io() {
        final Runnable task = mock(Runnable.class);
        mTaskExecutors.io(task);

        verify(mIo, times(1)).execute(task);
    }

    @Test
    public void terminate() {
        mTaskExecutors.terminate();

        verify(mCallback, times(1)).terminate();
        verify(mIo, times(1)).terminate();
    }
}

File: mmupnp/src/test/java/net/mm2d/upnp/internal/thread/TaskExecutorsTest.kt
/*
 * Copyright (c) 2019 大前良介 (OHMAE Ryosuke)
 *
 * This software is released under the MIT License.
 * http://opensource.org/licenses/MIT
 */

package net.mm2d.upnp.internal.thread

import io.mockk.mockk
import io.mockk.verify
import net.mm2d.upnp.TaskExecutor
import org.junit.Test
import org.junit.runner.RunWith
import org.junit.runners.JUnit4

@RunWith(JUnit4::class)
class TaskExecutorsTest {
    @Test
    fun callback() {
        val callbackTaskExecutor: TaskExecutor = mockk(relaxed = true)
        val ioTaskExecutor: TaskExecutor = mockk(relaxed = true)
        val taskExecutors = TaskExecutors(callbackTaskExecutor, ioTaskExecutor)
        val task: Runnable = mockk()
        taskExecutors.callback(task)

        verify(exactly = 1) { callbackTaskExecutor.execute(task) }
        taskExecutors.terminate()
    }

    @Test
    fun io() {
        val callbackTaskExecutor: TaskExecutor = mockk(relaxed = true)
        val ioTaskExecutor: TaskExecutor = mockk(relaxed = true)
        val taskExecutors = TaskExecutors(callbackTaskExecutor, ioTaskExecutor)
        val task: Runnable = mockk()
        taskExecutors.io(task)

        verify(exactly = 1) { ioTaskExecutor.execute(task) }
        taskExecutors.terminate()
    }

    @Test
    fun terminate() {
        val callbackTaskExecutor: TaskExecutor = mockk(relaxed = true)
        val ioTaskExecutor: TaskExecutor = mockk(relaxed = true)
        val taskExecutors = TaskExecutors(callbackTaskExecutor, ioTaskExecutor)
        taskExecutors.terminate()

        verify(exactly = 1) { callbackTaskExecutor.terminate() }
        verify(exactly = 1) { ioTaskExecutor.terminate() }
    }
}